bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/util/time/Interval.java

```diff
@@ -175,6 +176,28 @@ private boolean isCalendarBased() {
     return !getPeriod().isZero() || getDuration().compareTo(ACCURATE_DURATION_UPPER_BOUND) >= 0;
   }
 
+  /**
+   * Convert this interval in milliseconds
+   *
+   * @throws ArithmeticException in case of long overflow
+   * @return this interval converted to milliseconds
+   */
+  public long toTotalMilliseconds() {
+    final long durationMillis = getDuration().toMillis();
+    final Period innerPeriod = getPeriod();
+    final long periodMillis =
+        innerPeriod.getUnits().stream()
+            .map(
+                temporalUnit ->
+                    new Tuple<>(innerPeriod.get(temporalUnit), temporalUnit.getDuration()))
+            .mapToLong(
+                durationValuePair ->
+                    Math.multiplyExact(
+                        durationValuePair.getLeft(), durationValuePair.getRight().toMillis()))
+            .reduce(0, Math::addExact);
+    return Math.addExact(durationMillis, periodMillis);
```

Comment:
👍 Nice work coming up with a solution here

I thought about this code for some time, and I understand its need. Periods and Durations don't combine well, and Java's standard time library doesn't allow easily converting the two to milliseconds. The problem is that Periods may be mapped to different Durations depending on the TimeZone. With this in mind, I worried that this implementation might be incorrect in some situations. But does it matter for the current use case? It is useful to tell the engine to backoff a job for several days, but this does not have to be precise. So, in my opinion, it's fine to keep it this way. Let me know how you feel about this.

🔧 In any case, let's document this, so the method is not used for cases where this precision is required for Intervals containing Periods.